napari/layers/base/base.py

```diff
@@ -1594,9 +1595,51 @@ def _update_draw(
             corners[:, displayed_axes] = data_bbox_clipped
             self.corner_pixels = corners
 
-    def get_status(
+    def _get_source_info(self):
+        components = {}
+        if self.source.reader_plugin:
+            try:
```

Comment:
when would this throw `KeyError`? If `self.source.path` is `None`, `basename` throws `TypeError`?

Reply:
maybe we could avoid the try/except by doing 

```python
components['layer_base'] = os.path.basename(self.source.path or '')
```